streams/src/test/java/org/apache/kafka/streams/perf/YahooBenchmark.java

```diff
@@ -0,0 +1,375 @@
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.kstream.ForeachAction;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KStreamBuilder;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.KeyValueMapper;
+import org.apache.kafka.streams.kstream.Predicate;
+import org.apache.kafka.streams.kstream.TimeWindows;
+import org.apache.kafka.streams.kstream.ValueJoiner;
+import org.apache.kafka.streams.kstream.ValueMapper;
+import org.apache.kafka.streams.kstream.Windowed;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+
+
+/**
+ * A basic DSL and data generation that emulates the behavior of the Yahoo Benchmark
+ * https://yahooeng.tumblr.com/post/135321837876/benchmarking-streaming-computation-engines-at
+ * Thanks to Michael Armbrust for providing the initial code for this benchmark in his blog:
+ * https://databricks.com/blog/2017/06/06/simple-super-fast-streaming-engine-apache-spark.html
+ */
+public class YahooBenchmark {
+    private final SimpleBenchmark parent;
+    private final String campaignsTopic;
+    private final String eventsTopic;
+
+    static public class ProjectedEvent {
+        /* main fields */
+        public String eventType;
+        public String adID;
+
+        /* other fields */
+        public long eventTime;
+        public String userID;
+        public String pageID;
+        public String addType;
+        public String ipAddress;
+    }
+
+    static public class CampaignAd {
+        public String adID;
+        public String campaignID;
+    }
+
+    public YahooBenchmark(final SimpleBenchmark parent, final String campaignsTopic, final String eventsTopic) {
+        this.parent = parent;
+        this.campaignsTopic = campaignsTopic;
+        this.eventsTopic = eventsTopic;
+    }
+
+    // just for Yahoo benchmark
+    private boolean maybeSetupPhaseCampaigns(final String topic, final String clientId,
+                                             final boolean skipIfAllTests,
+                                             final int numCampaigns, final int adsPerCampaign,
+                                             final String[] ads) throws Exception {
+        parent.processedRecords.set(0);
+        parent.processedBytes = 0;
+        // initialize topics
+        if (parent.loadPhase) {
+            if (skipIfAllTests) {
+                // if we run all tests, the produce test will have already loaded the data
+                if (parent.testName.equals(SimpleBenchmark.ALL_TESTS)) {
+                    // Skipping loading phase since previously loaded
+                    return true;
+                }
+            }
+            System.out.println("Initializing topic " + topic);
+
+            parent.processedRecords.set(0);
```

Comment:
Doesn't look like next 2 lines are needed as already been set to 0 a few lines above.